Logical, big_integer and bound_integer generators and tests fixed

zofpolkowska opened the pull request.

Dzol commented on lib/orangeade/generator/boolean.ex:
```diff
@@ -0,0 +1,33 @@
+  """
+  def stream do
+    alias Orangeade.Generator.BoundNatural, as: BoundNatural
+    BoundNatural.stream(limit: 3)
```
I made a similar comment to Robert about piping as I would here... you decide what you consider better style.

Dzol commented on lib/orangeade/generator/boolean.ex:
```diff
@@ -0,0 +1,33 @@
+        e === 1 ->
+          :false
+        e === 2 ->
+          :nil
```
This is a great idea for a generator. I never considered `nil` alongside `true` and `false`. We need a different name for the generator though. `Boolean` doesn't fully capture the idea of a truthy or falsey value... the only other name I've come up with is `Logical` which isn't much of an improvement. Perhaps you have a better naming idea

Dzol commented on lib/orangeade/generator/printable_ascii.ex:
```diff
@@ -0,0 +1,35 @@
+    alias Psi.LinearCongruentialGenerator, as: LCG
+    
+    s = LCG.stream(generator: Orangeade.Generator.BigNatural.instance(), seed: 1)
+    Caffeine.Stream.map(s, fn e -> Integer.mod(e, printable()) + lower() end )
```
It's really nice to see a different approach

Dzol commented on test/orangeade_test.exs:
```diff
@@ -11,4 +13,17 @@ defmodule OrangeadeTest do
+
+  test "the boundnatural generator should be uniform" do
+    boundnaturals = Caffeine.Stream.take(BoundNatural.stream(limit: 100), 100)
+    epsilon = 5
```
The "epsilon" is great because it formally captures what you're doing. You can do it for the other test cases as well. It should be possible to abstract the relationship between `Stream.take`ing 100 and the `epsilon = 5`. I'm not sure how clean that'd be though.

Dzol commented on test/printable_ascii_test.exs:
```diff
@@ -0,0 +1,9 @@
+    s = Caffeine.Stream.take(                    
+      Orangeade.Generator.Printable.ASCII.stream(),
+      1_000)
+    assert List.ascii_printable?(s)
```
I had no idea this function existed!

Dzol commented on lib/orangeade/generator/bound_integer.ex:
```diff
@@ -0,0 +1,43 @@
+  [1, -12, 19, -110, 69, -120, 183, -18, 129, -44, 131, -54, 93, -48]
+
+  """
+  def stream(lower: i, upper: j) do
```
The `lower` and `upper` bounds are a nice addition to the API

zofpolkowska commented:
Hi,
Thanks for the comments! I fixed the module names for tests,
renamed my weird boolean to logical and added 'epsilon' where suitable (in my opinion).

Dzol commented on lib/orangeade/generator/printable_ascii.ex:
```diff
@@ -0,0 +1,35 @@
+defmodule Orangeade.Generator.Printable.ASCII do
```
This is the only remaining module name to change...

Dzol commented:
In case you want to re-introduce the tuple or function generators I updated the _Data Generators_ section in [class 3's notes](https://github.com/lambdaacademy/2018.03_elixir/blob/master/Class%20%233%20%E2%80%94%20Bound%20Integer%20%26%20Character%20List%20Generator.ipynb) with a more detailed description. I'll let you change the printable ASCII generator's module name and merge in the morning

zofpolkowska commented:
Thanks for the comments, I changed the name of printable_ascii to printable_ascii_character finally :smiley:

zofpolkowska commented:
I'm still not sure if it's done properly, but I added generators for tuple and function as well :)

Dzol commented:
@zofpolkowska If you resolve the conflicts above then I'll be able to merge. We can remove the modules from this PR or incorporate their differences. I don't mind which

zofpolkowska commented:
I removed them
